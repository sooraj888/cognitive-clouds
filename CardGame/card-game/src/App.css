.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}
input {
  min-width: 200px;
}
.grayCard {
  display: inline-block;
  background-color: gray;
  border: 1px solid white;
  min-width: 100px;
  min-height: 100px;
  max-width: 20%;
  max-height: 60%;
  /* text-indent: -9999px; */
}
.redCard {
  display: inline-block;
  background-color: red;
  border: 1px solid white;
  min-width: 100px;
  min-height: 100px;
  max-width: 20%;
  max-height: 60%;
  text-indent: 1;
  color: white;
}
.greenCard {
  display: inline-block;
  background-color: green;
  border: 1px solid white;
  color: white;
  min-width: 100px;
  min-height: 100px;
  max-width: 20%;
  max-height: 60%;
  text-indent: 1;
}
