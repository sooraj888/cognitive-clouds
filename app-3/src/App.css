.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
}
.listItems {
  display: inline-block;
  white-space: nowrap;
  max-width: 200px;
  min-width: 200px;
  text-overflow: ellipsis;
  overflow: hidden;
  padding: 5px 10px;
}
