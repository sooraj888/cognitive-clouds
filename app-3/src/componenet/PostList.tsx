import React, { useCallback, useEffect, useMemo, useState, memo } from "react";
import logo from "./logo.svg";

import Pagination from "@mui/material/Pagination";
import { Link } from "react-router-dom";

const PostList = ({
  countPost,
  postData,
  selectedPost,
  handleOnPaginationChange,
}: any) => {
  return (
    <div className="App">
      <h1 className="AppName">List</h1>
      <div className="list">
        <div className="listHead">
          <span className="listTitle"> Title </span>
          <span className="listTitle">URL</span>
          <span className="listTitle">Author</span>
          <span className="listTitle">Created-At</span>
          <span className="listTitle"></span>
        </div>

        {console.log(postData)}
        {postData && postData?.[selectedPost]
          ? postData?.[selectedPost].map((item: any) => {
              // console.log("item inside post data", item);

              return (
                <div key={item?.objectID} className="listItem">
                  <span className="listItems">{item?.title}</span>
                  <a className="listItems" href={item?.url}>
                    {item?.url}
                  </a>
                  <span className="listItems">{item.author} </span>
                  <span className="listItems">{item?.created_at}</span>
                  <span className="listItems">
                    <Link to={"/" + item?.objectID}>Select</Link>
                  </span>
                </div>
              );
            })
          : console.log("data not exits")}
      </div>
      <div className="paginationStyle">
        <Pagination
          variant="outlined"
          shape="rounded"
          count={countPost || 0}
          color="primary"
          onChange={handleOnPaginationChange}
          page={selectedPost + 1}
        />
      </div>
    </div>
  );
};

export default PostList;
