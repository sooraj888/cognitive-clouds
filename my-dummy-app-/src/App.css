.App {
  align-content: center;
  text-align: center;
}

.grayCard {
  display: inline-block;
  background-color: gray;
  border: 1px solid white;
  min-width: 100px;
  min-height: 100px;
  max-width: 20%;
  max-height: 60%;
  text-indent: 1;
}
.redCard {
  display: inline-block;
  background-color: red;
  border: 1px solid white;
  min-width: 100px;
  min-height: 100px;
  max-width: 20%;
  max-height: 60%;
  text-indent: 1;
}
.greenCard {
  display: inline-block;
  background-color: green;
  border: 1px solid white;
  min-width: 100px;
  min-height: 100px;
  max-width: 20%;
  max-height: 60%;
  text-indent: 1;
}
