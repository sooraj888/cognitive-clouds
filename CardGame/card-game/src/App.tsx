import React, { useEffect, useState } from "react";

import "./App.css";

function App() {
  const [isGameSceen, setIsGameSceen] = useState(false);
  const [isGameEnd, setIsGameEnd] = useState(false);
  const [playerName, setPlayerName] = useState("");
  const [enteredNumber, setEnterdNumber] = useState<number>(NaN);
  const [cards, setCards] = useState<any>([]);

  const arrayOfCards: any = [];
  const randomNumberCollector: any = [];

  let randomNumber: number = 0;

  const checkingRondomNumber = (randomNumber: number) => {
    let foundSameNumber = false;
    for (let k = 0; k < randomNumberCollector.length; k++) {
      if (randomNumberCollector[k] == randomNumber) {
        foundSameNumber = true;
      }
    }
    if (foundSameNumber) {
      return false;
    } else {
      return true;
    }
  };

  const genrateRandomArray = () => {
    for (let i = 0; i < 9; i++) {
      // //top for genrating random array
      // {
      //   for (let j = 0; j < randomNumberCollector.length; j++) {
      //     if (randomNumber === randomNumberCollector[j]) {
      //       randomNumber = 0;
      //     }
      //   }

      //
      // }
      // // randomNumber
      // // botom
      do {
        randomNumber = Math.floor(Math.random() * 9) + 1;
      } while (!checkingRondomNumber(randomNumber));

      randomNumberCollector[i] = randomNumber;

      arrayOfCards[i] = {
        id: i + 1,
        className: "grayCard",
        randomNumber: randomNumber,
      };
    }
    setCards(arrayOfCards);

    //random number array
    console.log("randomNumberCollector", randomNumberCollector);
  };

  const handleOnInputFormSubmit = (e: any) => {
    e.preventDefault();
    console.log("submited");
    setIsGameSceen(true);
    genrateRandomArray();
  };

  const handleOnNumberChange = (e: any) => {
    setEnterdNumber(parseInt(e.target.value));
  };
  const handleOnNameChange = (e: any) => {
    setPlayerName(e.target.value);
  };
  const handleExitOnClick = () => {
    setIsGameSceen(false);
    setPlayerName("");
    setEnterdNumber(NaN);
  };

  return (
    <div className="App">
      {!isGameSceen ? (
        <div>
          <h1>Card game</h1>
          <form onSubmit={handleOnInputFormSubmit}>
            <input
              value={playerName}
              type="text"
              autoComplete="on"
              onChange={handleOnNameChange}
              placeholder="Enter your name"
              required
            ></input>
            <br></br>
            <input
              value={enteredNumber || ""}
              type="number"
              onChange={handleOnNumberChange}
              required
              autoComplete="off"
              placeholder="Enter a number between 1 to 9"
              min="1"
              max="9"
            ></input>
            <br />
            <button type="submit">Start</button>
          </form>
        </div>
      ) : (
        <div>
          <>
            <h5>player Name : {playerName}</h5>
            <h5>Entered Number : {enteredNumber}</h5>

            <br></br>
            <hr></hr>
            {cards.map((cardItem: any) => {
              return (
                <button className={cardItem.className} key={cardItem.id}>
                  {cardItem.randomNumber}
                </button>
              );
            })}
          </>
          <br></br>
          {!isGameEnd ? (
            <></>
          ) : (
            <>
              <button>Retry</button>
              <button onClick={handleExitOnClick}>Exit</button>
            </>
          )}
        </div>
      )}
    </div>
  );
}

export default App;
