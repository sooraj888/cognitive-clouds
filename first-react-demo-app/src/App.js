import logo from './logo.svg';
import './App.css';
const title = "react"
const welcome = {
    greeting: 'Hey',
    title: 'React',
};
const list = [
    {
    title: 'React',
    url: 'https://reactjs.org/',
    author: 'Jordan Walke',
    num_comments: 3,
    points: 4,
    objectID: 0,
    },
    {
    title: 'Redux',
    url: 'https://redux.js.org/',
    author: 'Dan Abramov, Andrew Clark',
    num_comments: 2,
    points: 5,
    objectID: 1,
    },
    ];

function setTitle(titlename) {
    return (titlename);
}

function App() {
    
    return (
        <div>
        ...
        <hr/>
        <ul>
        {list.map(function (item) {
        return (<div>
                    <li>{item.title}</li>
                    <li>{item.url}</li>
                    <li>{item.author}</li>
                    <li>{item.num_comments}</li>
                    <li>{item.points}</li>
                    <br></br>
                    
                </div>);
        })}
        </ul>
        </div>);
       
   
}

export default App;