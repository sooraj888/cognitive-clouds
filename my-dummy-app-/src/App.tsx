import React from "react";
import logo from "./logo.svg";
import { useState } from "react";
import "./App.css";

function App() {
  const [searchtearm, setSearchTearm] = useState("aa");

  const handleSearch = (e: any) => {
    console.log(e.target.value);
    setSearchTearm(e.target.value);
  };

  return (
    <div className="App">
      <input value={searchtearm} onChange={handleSearch} type="text"></input>
    </div>
  );
}

export default App;
