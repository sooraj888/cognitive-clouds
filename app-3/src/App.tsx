import React, { useCallback, useEffect, useState } from "react";
import logo from "./logo.svg";
import "./App.css";
import Pagination from "@mui/material/Pagination";

import PostList from "./componenet/PostList";
import { BrowserRouter, Route, Routes } from "react-router-dom";

function App() {
  const [countPost, setCountPost] = useState<number>(NaN);
  const [postData, setPostData] = useState<any>([]);
  const [selectedPost, setSelectedPost] = useState<number>(0);

  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route
            path="/"
            element={
              <PostList
                countPost={countPost}
                setCountPost={setCountPost}
                postData={postData}
                setPostData={setPostData}
                selectedPost={selectedPost}
                setSelectedPost={setSelectedPost}
              ></PostList>
            }
          ></Route>

          {postData && postData?.[selectedPost]
            ? postData?.[selectedPost].map((item: any) => {
                return (
                  <Route
                    key={item?.objectID}
                    path={"/" + item?.objectID}
                    element={
                      <div className="rawJson">
                        <h1>Raw Json of selected post</h1>
                        {JSON.stringify(item)}
                      </div>
                    }
                  ></Route>
                );
              })
            : console.log("data not exits")}
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
