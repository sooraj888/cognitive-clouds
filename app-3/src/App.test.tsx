import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";
import PostList from "./componenet/PostList";
import { BrowserRouter, Routes } from "react-router-dom";

// countPost,
// postData,
// selectedPost,
// handleOnPaginationChange,

//title objectID url author created_at

const storyOne = {
  title: "Google",
  objectID: "1",
  url: "http://google.com",
  author: "Jimmy",
  created_at: "24-44-33",
};
const storyTwo = {
  title: "Microsoft",
  objectID: "2",
  url: "http://microsoft.com",
  author: "Max",
  created_at: "24-445-33",
};
const postData = [[storyOne, storyTwo]];

describe("test on PostList ", () => {
  test("displaying PostList componenet without sending dummy props", () => {
    render(<PostList></PostList>);
    screen.debug();
  });
});

describe(" PostList ", () => {
  test("displaying PostList componenet with dummy props ", () => {
    const PostListProps = {
      postData: postData,
      countPost: 1,
      selectedPost: 0,
      handleOnPaginationChange: jest.fn(),
    };

    render(
      <BrowserRouter>
        <PostList {...PostListProps}></PostList>
      </BrowserRouter>
    );

    screen.debug();
  });
});

describe(" PostList ", () => {
  const PostListProps = {
    postData: postData,
    countPost: 1,
    selectedPost: 0,
    handleOnPaginationChange: jest.fn(),
  };
  test("rendering all properties ", () => {
    render(
      <BrowserRouter>
        <PostList {...PostListProps}></PostList>
      </BrowserRouter>
    );
    expect(screen.getByText(/Microsoft/)).toBeInTheDocument();

    expect(screen.getByText("http://google.com")).toHaveAttribute(
      "href",
      "http://google.com"
    );
  });
});

describe(" PostList ", () => {
  const PostListProps = {
    postData: postData,
    countPost: 1,
    selectedPost: 0,
    handleOnPaginationChange: jest.fn(),
  };
  test("rendering a link ", () => {
    render(
      <BrowserRouter>
        <PostList {...PostListProps}></PostList>
      </BrowserRouter>
    );
    expect(screen.getByRole("navigation")).toBeInTheDocument();
  });
});

describe(" PostList ", () => {
  const PostListProps = {
    postData: postData,
    countPost: 1,
    selectedPost: 0,
    handleOnPaginationChange: jest.fn(),
  };
  test("clicking first pagination button", () => {
    render(
      <BrowserRouter>
        <PostList {...PostListProps}></PostList>
      </BrowserRouter>
    );

    fireEvent.click(screen.getByText("1"));
    expect(PostListProps.handleOnPaginationChange).toHaveBeenCalledTimes(1);
    screen.debug();
  });
});

describe(" PostList ", () => {
  const PostListProps = {
    postData: postData,
    countPost: 1,
    selectedPost: 0,
    handleOnPaginationChange: jest.fn(),
  };
  test("count of select link ", () => {
    render(
      <BrowserRouter>
        <PostList {...PostListProps}></PostList>
      </BrowserRouter>
    );

    expect(screen.getAllByText(/Select/).length).toBe(2);
  });
});

describe(" PostList ", () => {
  const PostListProps = {
    postData: postData,
    countPost: 2,
    selectedPost: 0,
    handleOnPaginationChange: jest.fn(),
  };
  test("renders a snapshot ", () => {
    const { container } = render(
      <BrowserRouter>
        <PostList {...PostListProps}></PostList>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
