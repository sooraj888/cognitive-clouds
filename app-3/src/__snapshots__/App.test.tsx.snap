// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` PostList  renders a snapshot  1`] = `
<div>
  <div
    class="App"
  >
    <h1
      class="AppName"
    >
      List
    </h1>
    <div
      class="list"
    >
      <div
        class="listHead"
      >
        <span
          class="listTitle"
        >
           Title 
        </span>
        <span
          class="listTitle"
        >
          URL
        </span>
        <span
          class="listTitle"
        >
          Author
        </span>
        <span
          class="listTitle"
        >
          Created-At
        </span>
        <span
          class="listTitle"
        />
      </div>
      <div
        class="listItem"
      >
        <span
          class="listItems"
        >
          Google
        </span>
        <a
          class="listItems"
          href="http://google.com"
        >
          http://google.com
        </a>
        <span
          class="listItems"
        >
          Jimmy
           
        </span>
        <span
          class="listItems"
        >
          24-44-33
        </span>
        <span
          class="listItems"
        >
          <a
            href="/1"
          >
            Select
          </a>
        </span>
      </div>
      <div
        class="listItem"
      >
        <span
          class="listItems"
        >
          Microsoft
        </span>
        <a
          class="listItems"
          href="http://microsoft.com"
        >
          http://microsoft.com
        </a>
        <span
          class="listItems"
        >
          Max
           
        </span>
        <span
          class="listItems"
        >
          24-445-33
        </span>
        <span
          class="listItems"
        >
          <a
            href="/2"
          >
            Select
          </a>
        </span>
      </div>
    </div>
    <div
      class="paginationStyle"
    >
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root MuiPagination-outlined css-1oj2twp-MuiPagination-root"
      >
        <ul
          class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-disabled MuiPaginationItem-previousNext css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-current="true"
              aria-label="page 1"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-selected MuiPaginationItem-page css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              1
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 2"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-page css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              2
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary MuiPaginationItem-previousNext css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
